docs/GUIDE.md

```diff
@@ -0,0 +1,142 @@
+- `--disable-detection-generic-methods`
+
+This option disables detection of messages/events/state based on their usage as generic argument to a method, e.g. `akka.actor.typed.scaladsl.ActorContext.ask`. It is disabled by default. If you want to use it, add following setting:<br>
+`Compile / scalacOptions += "-P:serializability-checker-plugin:--disable-detection-generic-methods"`<br><br>
+- `--disable-detection-methods`
+
+This option disables detection of messages/events/state based on type of arguments to a method, e.g. `akka.actor.typed.ActorRef.tell`. It is disabled by default. If you want to use it, add following setting:<br>
+`Compile / scalacOptions += "-P:serializability-checker-plugin:--disable-detection-methods"`<br><br>
+- `--disable-detection-untyped-methods`
+
+This option disables detection of messages/events/state based on type of arguments to a method that takes Any, used for Akka Classic. It is disabled by default. If you want to use it, add following setting:<br>
+`Compile / scalacOptions += "-P:serializability-checker-plugin:--disable-detection-untyped-methods"`<br><br>
+- `--disable-detection-higher-order-function`
+
+This option disables detection of messages/events/state based on return type of the function given as argument to method. It is disabled by default. If you want to use it, add following setting:<br>
+`Compile / scalacOptions += "-P:serializability-checker-plugin:--disable-detection-higher-order-function"`<br><br>
+
+### Codec Registration Checker Compiler Plugin
+Before using this compiler plugin, make sure that you are using both [annotations](#annotations) properly. If yes &mdash; plugin can be used right away. This plugin checks whether classes marked with serializability trait are being referenced in a marked serializer, which ensures that codecs will be registered in runtime.<br><br>
+Codec Registration Checker Compiler Plugin does not need additional configuration, but you can change default configurations as explained below:
+- `--disable`
+
+The Plugin is enabled by default in each project / module where `AkkaSerializationHelperPlugin` is enabled. If you want to disable this particular plugin (but want to keep on using other ASH compiler plugins) &mdash; add following setting inside `build.sbt` for selected project / module:<br>
+`Compile / scalacOptions += "-P:codec-registration-checker-plugin:--disable"`<br><br>
+- `custom cache file path`
+
+This plugin uses a helper cache file named `codec-registration-checker-cache.csv`. The file is created under `target/` directory by default, so that it will not bother users. This file is useful if user performs incremental compilation. If you want, you can specify your custom path instead of the default one. To do so, add following configuration to the plugin:<br>
+`Compile / scalacOptions += "-P:codec-registration-checker-plugin:PATH_TO_CUSTOM_DIRECTORY"`<br><br>
```

Comment:
Okay... so finally we pass the directory as a loose argument and NOT as a named flag right?

Reply:
Yes, this has not been changed in deed. IMO we should switch to named flag. Issue for this change created: https://github.com/VirtusLab/akka-serialization-helper/issues/210

Reply:
Moreover - the same applies to the dump persistence schema compiler plugin. Also mentioned in #210 + I added similar description to this plugin in the guide (I must have missed it earlier)